File: 02 tests/ex7_object_ops.js
// run with: deno run ex7_object_ops.js | node ex7_object_ops.js

/* ex7_object_ops.js

Object manipulation

:author:  wolf
:created: 2025.05.22
*/

const iterations = 3;
const timing_results = [];

// define the test function
// --- function to be timed-- -

function jsObjectOps(n, m) {
    const obj = {};
    for (let i = 0; i < n; i++) {
        obj[i] = Math.floor(Math.random() * 101);
    }
    let total = 0;
    for (let i = 0; i < m; i++) {
        total += obj[Math.floor(Math.random() * n)];
    }
    return total;
}


for (let i = 0; i < iterations; i++) {
    const start = performance.now();

    // call the test function
    jsObjectOps(10**7, 10**5);

    const end = performance.now();

    timing_results.push(end - start);
}


console.log("Elapsed time: ",
    timing_results.reduce((a, b) => a + b) / iterations / 1000, "seconds");

File: 02 tests/ex7_object_ops.py
#!/usr/bin/env python3

""" ex7_object_ops.py

Object manipulation

:author:	wolf
:created:	2025.05.22
"""

import time
import random

iterations = 3
timing_results = []

# define the test function
# --- function to be timed ---


def python_dict_ops(n, m):
    d = {}
    for i in range(n):
        d[str(i)] = random.randint(0, 100)
    total = 0
    for _ in range(m):
        total += d[str(random.randint(0, n-1))]
    return total


for i in range(iterations):
    start = time.perf_counter()

    # call the test function
    python_dict_ops(10**7, 10**5)

    end = time.perf_counter()

    timing_results.append(end - start)


print(f"Elapsed time:  {sum(timing_results)/iterations:.16f} seconds")
